lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -37,6 +44,27 @@ def serializable_hash(options = {})
 
         private
 
+        def remove_duplicates(hash)
+          if @options[:prevent_duplicates] && hash[:included] && hash[:data].is_a?(Array)
```

Comment:
I think, the first one would increase readability, the second one would reduce the nesting ;-) 

take the second one.